### DISCUSSION TITLE
Is there a way to download achievements from the website?

### DISCUSSION THREAD

**Discussion by mainrs** [RetroAchievements/RAIntegration#907] [Q&A]
I have been working on a game specific DSLs to create achievements. Instead of directly working with binary data I abstract it away and provide means to access stats.

The prototype works for Pokemon Red. I can query for conditions like `receives item MasterBall`, `receives special item Pokedex` or `caught 5 water pokemon` without having to work with the memory directly.

I was wondering if there is a way to download achievements from the retroachievements.org that I can port over to my DSL and see if my implementation works properly.

I cannot find any repository containing the achievements. And there does not seem to be a download option on the website either.

**Comment by Jamiras** [comment]
Achievement definitions only exist in the database. There's no repository to download them from.

If you load the game in any of the standalone emulators, the data (including the achievement definitions) is stored in the `RACache\Data\XXX.json file`, where XXX is the game ID.

You might also be interested in looking at [RATools](https://github.com/Jamiras/RATools/wiki), which already provides a [DSL abstraction](https://github.com/Jamiras/RATools/wiki/Script-File-Syntax) for creating achievements.

```
achievement(
    title = "Light to 500",
    description = "Get at least 500 points in Count Up using light darts",
    points = 5,
    trigger = game_variant() == variant_countup &&
              player_is_player(0) && 
              current_player() == 1 && 
              byte(0x04F2) >= 5 &&                       // score (hundreds)
              throwing_phase() == phase_hit_board && 
              player_dart_weight(0) == weight_light &&
              never_using_alien_aim()
)
```
